modules/ui/fields/address.js

```diff
@@ -36,91 +36,94 @@ export function uiFieldAddress(field, context) {
             .filter(isAddressable)
-            .map(function(d) {
-                var loc = context.projection([
-                    (extent[0][0] + extent[1][0]) / 2,
-                    (extent[0][1] + extent[1][1]) / 2
-                ]);
-                var choice = geoChooseEdge(context.graph().childNodes(d), loc, context.projection);
-
+            .map(d => {
+                let dist = geoSphericalDistance(d.extent(context.graph()).center(), l);
+
+                if (d.type === 'way') {
+                    var loc = context.projection([
+                        (extent[0][0] + extent[1][0]) / 2,
+                        (extent[0][1] + extent[1][1]) / 2
+                    ]);
+                    var choice = geoChooseEdge(context.graph().childNodes(d), loc, context.projection);
+                    dist = Math.min(dist, choice.distance);
+                }
+
+                const value = resultProp && d.tags[resultProp] ? d.tags[resultProp] : d.tags.name;
+                let title = value;
+                if (type === 'street') {
+                    title = `${addrField.t('placeholders.street')}: ${title}`;
+                } else if (type === 'place') {
+                    title = `${addrField.t('placeholders.place')}: ${title}`;
+                }
                 return {
-                    title: d.tags.name,
-                    value: d.tags.name,
-                    dist: choice.distance
+                    title,
+                    value,
+                    dist,
+                    type,
+                    klass: `address-${type}`
                 };
             })
             .sort(function(a, b) {
                 return a.dist - b.dist;
             });
 
-        return utilArrayUniqBy(streets, 'value');
+        return utilArrayUniqBy(features, 'value');
+    }
 
+    function getNearStreets() {
         function isAddressable(d) {
             return d.tags.highway && d.tags.name && d.type === 'way';
         }
-    }
-
 
-    function getNearCities() {
-        var extent = combinedEntityExtent();
-        var l = extent.center();
-        var box = geoExtent(l).padByMeters(200);
-
-        var cities = context.history().intersects(box)
-            .filter(isAddressable)
-            .map(function(d) {
-                return {
-                    title: d.tags['addr:city'] || d.tags.name,
-                    value: d.tags['addr:city'] || d.tags.name,
-                    dist: geoSphericalDistance(d.extent(context.graph()).center(), l)
-                };
-            })
-            .sort(function(a, b) {
-                return a.dist - b.dist;
-            });
+        return getNear(isAddressable, 'street', 200);
+    }
 
-        return utilArrayUniqBy(cities, 'value');
+    function getNearPlaces() {
+        function isAddressable(d) {
+            return d.tags.name && d.tags.place ||
+                d.tags.boundary === 'administrative' && d.tags.admin_level >= 8;
```

Comment:
For the countries that normally use `addr:place`, is it likely to be the kind of place that has an administrative boundary relation, or something less formal that’s only mapped as a point feature? Could the criteria vary by country?

Reply:
From my experience, _small places_ are more common, but there are also cases where they reference an admin boundary (example: https://www.openstreetmap.org/node/296611194 -> https://www.openstreetmap.org/relation/436341). Perhaps referencing an `admin_level=8` boundary would be a bit too "big" of an object to reference, though. And offering these suggestions could in worst case potentially even lead to users mistakenly choosing this field instead of the intended `addr:city`, which would not be great. I'm changing it to `admin_level>8`.